src/Task/SecurityCheckerRoave.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Formatter\ProcessFormatterInterface;
+use GrumPHP\Process\ProcessBuilder;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use GrumPHP\Util\Filesystem;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class SecurityCheckerRoave extends AbstractExternalTask
+{
+    /** @var Filesystem */
+    private $filesystem;
+
+    public function __construct(
+        ProcessBuilder $processBuilder,
+        ProcessFormatterInterface $formatter,
+        Filesystem $filesystem
+    ) {
+        parent::__construct($processBuilder, $formatter);
+        $this->filesystem = $filesystem;
+    }
+
+    public static function getConfigurableOptions(): OptionsResolver
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'jsonfile' => './composer.json',
+            'lockfile' => './composer.lock',
+            'run_always' => false,
+        ]);
+
+        $resolver->addAllowedTypes('jsonfile', ['string']);
+        $resolver->addAllowedTypes('lockfile', ['string']);
+        $resolver->addAllowedTypes('run_always', ['bool']);
+
+        return $resolver;
+    }
+
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return $context instanceof GitPreCommitContext || $context instanceof RunContext;
+    }
+
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+
+        $config = $this->getConfig()->getOptions();
+        $composerFile = $config['jsonfile'];
+
+        if (!$this->filesystem->isFile($composerFile)) {
+             return TaskResult::createSkipped($this, $context);
+        }
+
+        if (!$this->hasRoaveSecurityAdvisoriesInstalled($composerFile)) {
+            return TaskResult::createFailed(
+                $this,
+                $context,
+                'This task is only available when roave/security-advisories is installed as a library.'
+            );
+        }
+
+        $config = $this->getConfig()->getOptions();
+        $files = $context->getFiles()
+            ->path(pathinfo($config['lockfile'], PATHINFO_DIRNAME))
+            ->name(pathinfo($config['lockfile'], PATHINFO_BASENAME));
+        if (0 === \count($files) && !$config['run_always']) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('composer');
```

Comment:
Wouldnt this be very slow in comparison to the other tasks?
I can imagine building the dependency tree takes up a lot of memory and time on larger projects?